packages/engine/src/datastore/table/context/mod.rs

```diff
@@ -89,37 +86,100 @@ impl Context {
     pub fn into_pre_context(self) -> PreContext {
         PreContext {
             batch: self.batch,
-            local_meta: self.local_meta,
+            removed_batches: self.removed_batches,
         }
     }
+
+    /// Copies the current agent `State` into the `Context` before running state packages, which
+    /// stores a snapshot of state at the end of the last step.
```

Comment:
Perhaps we should call it `store_state_snapshot`, `update_state_snapshot`, or `refresh_state_snapshot` or something instead of `synchronize_agent_pool`?

@litvand probably has an opinion